compiler/main.nim

```diff
@@ -186,6 +177,37 @@ proc commandView(graph: ModuleGraph) =
 const
   PrintRopeCacheStats = false
 
+proc hashMainCompilationParams*(conf: ConfigRef): string =
+  ## doesn't have to be complete; worst case results in a cache hit and a
+  ## recompilation.
+  var state = newSha1State()
+  macro updateState(state, body) =
+    result = newStmtList()
+    for ai in body:
+      result.add quote do:
+        `state`.update($`ai`)
+  updateState(state):
+    os.getAppFilename() # nim compiler
+    conf.commandLine # excludes `arguments`, as it should
+    conf.projectFull
```

Comment:
```suggestion
  state.update os.getAppFilename() # nim compiler
  state.update conf.commandLine # excludes `arguments`, as it should
  state.update conf.projectFull
```

Or maybe use the `with` macro or whatever it's called.

Reply:
done